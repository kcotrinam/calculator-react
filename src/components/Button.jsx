import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ name, handlerClick }) => {
  const handleClick = () => handlerClick(name);

  return (
    <button
      onClick={handleClick}
      type="button"
    >
      {name}
    </button>
  );
};

Button.propTypes = {
  name: PropTypes.string,
  handlerClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  name: '',
};

export default Button;
