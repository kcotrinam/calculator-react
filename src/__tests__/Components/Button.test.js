import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import Button from '../../components/Button';

test('renders correctly', () => {
  const tree = create(<Button handlerClick={() => 'hello'} name="name" />).toJSON();
  expect(tree).toMatchSnapshot();
});
