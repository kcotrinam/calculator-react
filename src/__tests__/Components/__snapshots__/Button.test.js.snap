// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
  name
</button>
`;
